src/seq/abif/reader.jl

```diff
@@ -0,0 +1,254 @@
+        data_offset = position(input)
+    else
+        data_offset = ntoh(first(read(input, Int32, 1)))
+    end
+
+    AbifDirEntry(name, number, element_type, element_size, num_elements, data_size, data_offset)
+end
+
+# read bytes according to the element type, other values are unsupported or legacy.
+function parse_data_tag(a::AbifReader, tag::AbifDirEntry)
+    if tag.data_offset > 0
+        seek(a.input, tag.data_offset)
+
+        if tag.element_type == 2
+            data = String(read(a.input, tag.data_size))
+
+            return format_data!(data, tag, elements(a, tag.name))
+        elseif tag.element_type == 4
+            data = read(a.input, Int16, tag.num_elements)
+            data = convert_to_int!(data)
+
+            return format_data!(data, tag, elements(a, tag.name))
+        elseif tag.element_type == 5
+            data = read(a.input, Int32, tag.num_elements)
+            data = convert_to_int!(data)
+
+            return format_data!(data, tag, elements(a, tag.name))
+        elseif tag.element_type == 7
+            data = read(a.input, Float32, tag.num_elements)
+            data = convert_to_float!(data)
+
+            return format_data!(data, tag, elements(a, tag.name))
+        elseif tag.element_type == 10
+            year   = Int(ntoh(first(read(a.input, Int16, 1))))
+            month  = Int(first(read(a.input, UInt8, 1)))
+            day    = Int(first(read(a.input, UInt8, 1)))
+            data   = "$year-$month-$day"
+
+            return format_data!(data, tag, elements(a, tag.name))
+        elseif tag.element_type == 11
+            hour    = Int(ntoh(first(read(a.input, 1))))
+            minute  = Int(ntoh(first(read(a.input, 1))))
+            second  = Int(ntoh(first(read(a.input, 1))))
+            hsecond = Int(ntoh(first(read(a.input, 1))))
+            data    = "$hour:$minute:$second:$hsecond"
+
+            return format_data!(data, tag, elements(a, tag.name))
+        elseif tag.element_type == 13
+            data = Bool(first(read(a.input, tag.data_size)))
+
+            return format_data!(data, tag, elements(a, tag.name))
+        elseif tag.element_type == 18
+            data = String(read(a.input, tag.data_size)[2:end])
+
+            return format_data!(data, tag, elements(a, tag.name))
+        elseif tag.element_type == 19
+            data = String(read(a.input, tag.data_size)[1:end-1])
+
+            return format_data!(data, tag, elements(a, tag.name))
+        else
+            data = read(a.input, tag.data_size)
+            return format_data!(data, tag, elements(a, tag.name))
+        end
+    end
+end
+
+# if TAG has more than one element, concatenate element number on name
+function format_data!(data::Any, tag::AbifDirEntry, elements::Int)
+
+    if elements > 1
+        return ("$(tag.name)$(tag.number)", data)
+    end
+
+    return ("$(tag.name)", data)
+end
+
+# convert a array of big-endian values
+function convert_to_int!(data::AbstractArray)
```

Comment:
Function name ending with `!` means a modifying function in Julia, which will modify the argument(s) passed to the function. Here, it is not the case, so please consider changing the name.